serving/deploying-with-private-registry.md

```diff
@@ -0,0 +1,211 @@
+# Deploying to Knative using a private container registry
```

Comment:
How about update the title and file name as `Deploying to Knative using a private container registry on IBM Cloud`?

Reply:
I was hoping that the instructions were general purpose enough to be useful for other providers. The part I originally tripped up that motivated me to create this doc was creating the correct secrets. @samodell thoughts?

Reply:
I'm happy with the current name. Up to you, @beemarie !

Reply:
I'll plan to keep it as is, and if we add another provider we can change the name to differentiate.